manifests/producer/config.pp

```diff
@@ -8,23 +8,25 @@
 # It manages the producer config files
 #
 class kafka::producer::config(
-  $config          = {},
+  $config          = $kafka::producer::config,
+  $config_defaults = $kafka::producer::config_defaults,
   $service_restart = $kafka::producer::service_restart
 ) {
 
   $producer_config = deep_merge($kafka::params::producer_config_defaults, $config)
 
   $config_notify = $service_restart ? {
-    true    => Service['kafka'],
+    true    => Service['kafka-producer'],
     default => undef
   }
 
   file { '/opt/kafka/config/producer.properties':
     ensure  => present,
```

Comment:
This is not necessary and we do not use it for our other modules. 
https://docs.puppetlabs.com/puppet/latest/reference/type.html#file-attribute-ensure